tasks/internal/push-artifacts-to-cdn-task/push-artifacts-to-cdn-task.yaml

```diff
@@ -145,7 +222,558 @@ spec:
+          mountPath: /shared
+      env:
+        - name: QUAY_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: username
+        - name: QUAY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: password
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        STDERR_FILE=/tmp/stderr.txt
+
+        # Check if the previous step finished successfully. If not, stop here.
+        if [ "$(cat "$(results.result.path)")" != "Success" ]; then
+          echo "Previous step failed. Exiting..."
+          exit 0
+        fi
+
+        exitfunc() {
+            local err=$1
+            local line=$2
+            local command="$3"
+            if [ "$err" -eq 0 ] ; then
+                echo -n "Success" > "$(results.result.path)"
+            else
+                echo "$0: ERROR '$command' failed at line $line - exited with status $err" \
+                  > "$(results.result.path)"
+                if [ -f "$STDERR_FILE" ] ; then
+                    tail -n 20 "$STDERR_FILE" >> "$(results.result.path)"
+                fi
+            fi
+            exit 0 # exit the script cleanly as there is no point in proceeding past an error or exit call
+        }
+        # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        CONTENT_DIR=/shared/artifacts
+        find $CONTENT_DIR
+        UNSIGNED_DIR=$CONTENT_DIR/unsigned
+        MAC_CONTENT=$UNSIGNED_DIR/macos
+        WINDOWS_CONTENT=$UNSIGNED_DIR/windows
+        LINUX_CONTENT=$CONTENT_DIR/linux
+
+        mkdir -p "$MAC_CONTENT" "$WINDOWS_CONTENT" "$LINUX_CONTENT"
+        cd "$CONTENT_DIR"
+
+        # Loop through each gz file and move them to the appropriate directory
+        find "$CONTENT_DIR" -type f -iregex ".*\.zip\|.*\.gz" 2>/dev/null | while read -r file; do
+          case "$file" in
+            (*darwin*)
+              mv "$file" unsigned/macos/
+              ;;
+            (*windows*)
+              mv "$file" unsigned/windows/
+              ;;
+            (*linux*)
+              mv "$file" linux/
+              ;;
+          esac
+        done
+
+        # Unzip files in each directory
+        for dir in unsigned/macos unsigned/windows linux; do
+          find "$dir"/*.{zip,gz} 2>/dev/null | while read -r file; do
+            ext="${file#*.}"
+            case "$ext" in
+              tar.gz)
+                tar -xf "$file"
+              ;;
+              gz)
+                gzip -d "$file"
+              ;;
+              zip)
+                zip -X "$file"
+              ;;
+            esac
+          done
+        done
+
+        cd "$UNSIGNED_DIR"
+
+        echo "Logging into Quay..."
+        #set +x
+        oras login quay.io -u "${QUAY_USER}" -p "${QUAY_PASS}" #> /dev/null 2>&1
```

Comment:
The `#` look like an omission from testing?

Reply:
fixed.